src/voice_generator/__main__.py

```
@@ -0,0 +1,248 @@
+                audio = forward_gpu(ps, ref_s, speed)
+            else:
+                audio = models[False](ps, ref_s, speed)
+        except gr.exceptions.Error as e:
+            if use_gpu:
+                gr.Warning(str(e))
+                gr.Info(
+                    "Retrying with CPU. To avoid this error, change Hardware to CPU."
+                )
+                audio = models[False](ps, ref_s, speed)
+            else:
+                raise gr.Error(e.message)
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)